src/reactrails_in_reagent/core.cljs

```diff
@@ -0,0 +1,118 @@
+(ns reactrails-in-reagent.core
+  (:require [reagent.core :as reagent :refer [atom]]))
+
+(enable-console-print!)
+
+(println "Edits to this text should show up in your developer console.")
+
+;; define your app data so that it doesn't get over-written on reload
+
+(def app-state (atom {:author ""
+                      :nav 0
+                      :comments [{:author "Bob" :text "The Price Is Right!"}] }))
+
+(defn add-comment! [c]
+  (swap! app-state update :comments (comp reverse   #(conj % )) c )
+
+(defn handle-submit [e]
+  (.preventDefault e)
+  (let [author (.-value (.getElementById js/document   "author"))
+        text (.-value (.getElementById js/document "text"))]
+    (aset  (.getElementById js/document "text") "value" "") ;resets textaraa
+    (add-comment! {:author author :text text}))))
+
+(defn handle-nav-change [index]
+  (swap! app-state assoc :nav index))
+
+(defn handle-author-change [e]
+  (let [author (-> e .-target .-value)]
+    (swap! app-state assoc :author author)))
+
+(defn Form-horizontal []
+  [:form {:class "commentForm form-horizontal" :onSubmit handle-submit}
```

Comment:
The ugliest part is where you interop with the javascript objects. Not a great way around that. I find the dsl for the html to be quite nice.